File: 20220803/모범답안/20220803_01.js/ReverseText_test.js
import assert from 'assert'
import solution from './ReverseText.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const s = 'apple'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('elppa', result)
    done();
  });

  it('test 2', function (done) {
    // given
    const s = 'ApPlE'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('ElPpA', result)
    done();
  });

  it('test 3', function (done) {
    // given
    const s = 'Orange'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('egnarO', result)
    done();
  });

  it('test 4', function (done) {
    // given
    const s = ''

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('', result)
    done();
  });

  it('test 5', function (done) {
    // given
    const s = 'A'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('A', result)
    done();
  });

  it('test 6', function (done) {
    // given
    const s = 'z'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('z', result)
    done();
  });
})

File: 20220803/모범답안/20220803_01.js/ReverseText_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.ReverseText.ReverseText import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        s = 'apple'

        # when
        result = solution(s)

        # then
        self.assertEqual('elppa', result)

    def test_2(self):
        # given
        s = 'ApPlE'

        # when
        result = solution(s)

        # then
        self.assertEqual('ElPpA', result)

    def test_3(self):
        # given
        s = 'Orange'

        # when
        result = solution(s)

        # then
        self.assertEqual('egnarO', result)

    def test_4(self):
        # given
        s = ''

        # when
        result = solution(s)

        # then
        self.assertEqual('', result)

    def test_5(self):
        # given
        s = 'A'

        # when
        result = solution(s)

        # then
        self.assertEqual('A', result)

    def test_6(self):
        # given
        s = 'z'

        # when
        result = solution(s)

        # then
        self.assertEqual('z', result)


if __name__ == '__main__':
    unittest.main()
